Add dropEffect getter and setter on DataTransfer

fehrenbach opened the pull request.

garyb commented on src/DOM/HTML/Event/DataTransfer.purs:
```diff
@@ -61,3 +66,27 @@ setData
+    "copy" -> Copy
+    "move" -> Move
+    "link" -> Link
+    "none" -> None
```
`unsafePartialBecause` is problematic (https://github.com/purescript/purescript-partial/issues/7), could you use `crashWith` in an `_` case, and then `unsafePartial` the `case` instead please?

garyb commented on src/DOM/HTML/Event/DataTransfer.purs:
```diff
@@ -61,3 +66,27 @@ setData
+
+foreign import dropEffectImpl :: forall eff. DataTransfer -> Eff (dom :: DOM | eff) String
+
+data DropEffect = None | Copy | Move | Link
```
Could you add derived instances for `Eq` and `Ord` for this too please?

fehrenbach commented:
No worries. What kind of heathen makes pull requests on Christmas Day anyways? ;)

Thanks for the review. I addressed your remarks and sorted the cases while I was at it.

garyb commented:
Thanks!
